File: app/src/main/java/com/cheatdatabase/activity/ui/mycheats/MyUnpublishedCheatsListViewAdapter.java
package com.cheatdatabase.activity.ui.mycheats;

import android.app.Activity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;

import com.cheatdatabase.R;
import com.cheatdatabase.data.model.UnpublishedCheat;
import com.cheatdatabase.listeners.MyUnpublishedCheatsListItemSelectedListener;

import java.util.ArrayList;
import java.util.List;

public class MyUnpublishedCheatsListViewAdapter extends RecyclerView.Adapter<MyUnpublishedCheatsListViewItemHolder> {
    private Activity activity;
    private List<UnpublishedCheat> unpublishedCheats;
    private MyUnpublishedCheatsListItemSelectedListener myUnpublishedCheatsListItemSelectedListener;

    public MyUnpublishedCheatsListViewAdapter(MyUnpublishedCheatsListItemSelectedListener myUnpublishedCheatsListItemSelectedListener, Activity activity) {
        unpublishedCheats = new ArrayList<>();
        this.activity = activity;

        this.myUnpublishedCheatsListItemSelectedListener = myUnpublishedCheatsListItemSelectedListener;
    }

    @Override
    public MyUnpublishedCheatsListViewItemHolder onCreateViewHolder(final ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.unpublished_cheat_list_item, parent, false);
        return new MyUnpublishedCheatsListViewItemHolder(itemView, activity);
    }

    public void onBindViewHolder(MyUnpublishedCheatsListViewItemHolder myUnpublishedCheatsListViewItemHolder, final int position) {
        myUnpublishedCheatsListViewItemHolder.updateUI(unpublishedCheats.get(position));
        myUnpublishedCheatsListViewItemHolder.detailsButton.setOnClickListener(v -> myUnpublishedCheatsListItemSelectedListener.onRejectReasonButtonClicked(unpublishedCheats.get(position)));
        myUnpublishedCheatsListViewItemHolder.editButton.setOnClickListener(v -> myUnpublishedCheatsListItemSelectedListener.onEditCheatButtonClicked(unpublishedCheats.get(position)));
        myUnpublishedCheatsListViewItemHolder.deleteButton.setOnClickListener(v -> myUnpublishedCheatsListItemSelectedListener.onDeleteButtonClicked(unpublishedCheats.get(position)));
    }

    public void setUnpublishedCheats(List<UnpublishedCheat> unpublishedCheats) {
        this.unpublishedCheats = unpublishedCheats;
    }

    @Override
    public int getItemCount() {
        return unpublishedCheats.size();
    }

}

File: app/src/main/java/com/cheatdatabase/activity/ui/mycheats/MyUnpublishedCheatsListViewAdapter.kt
package com.cheatdatabase.activity.ui.mycheats

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.cheatdatabase.data.model.UnpublishedCheat
import com.cheatdatabase.databinding.UnpublishedCheatListItemBinding
import com.cheatdatabase.listeners.MyUnpublishedCheatsListItemSelectedListener

class MyUnpublishedCheatsListViewAdapter(
    val myUnpublishedCheatsListItemSelectedListener: MyUnpublishedCheatsListItemSelectedListener,
    val activity: MyUnpublishedCheatsListActivity
) : RecyclerView.Adapter<MyUnpublishedCheatsListViewItemHolder>() {
    var unpublishedCheats: List<UnpublishedCheat> = ArrayList()

    private var _binding: UnpublishedCheatListItemBinding? = null
    private val binding get() = _binding!!

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): MyUnpublishedCheatsListViewItemHolder {
        _binding = UnpublishedCheatListItemBinding.inflate(
            LayoutInflater.from(parent.context),
            parent,
            false
        )

        return MyUnpublishedCheatsListViewItemHolder(binding, activity)
    }

    override fun onBindViewHolder(
        myUnpublishedCheatsListViewItemHolder: MyUnpublishedCheatsListViewItemHolder,
        position: Int
    ) {
        myUnpublishedCheatsListViewItemHolder.setCheat(unpublishedCheats[position])
        myUnpublishedCheatsListViewItemHolder.detailsButton.setOnClickListener {
            myUnpublishedCheatsListItemSelectedListener.onRejectReasonButtonClicked(
                unpublishedCheats[position]
            )
        }
        myUnpublishedCheatsListViewItemHolder.editButton.setOnClickListener {
            myUnpublishedCheatsListItemSelectedListener.onEditCheatButtonClicked(
                unpublishedCheats[position]
            )
        }
        myUnpublishedCheatsListViewItemHolder.deleteButton.setOnClickListener {
            myUnpublishedCheatsListItemSelectedListener.onDeleteButtonClicked(
                unpublishedCheats[position],
                position
            )
        }
    }

    override fun getItemCount(): Int {
        return unpublishedCheats.size
    }


}